components/esp32/pm_esp32.c

```diff
@@ -554,6 +565,9 @@ void IRAM_ATTR vApplicationSleep( TickType_t xExpectedIdleTime )
         }
     }
     portEXIT_CRITICAL(&s_switch_lock);
+    if (went_to_sleep && s_light_sleep_wakeup_cb != NULL) {
+        s_light_sleep_wakeup_cb();
```

Comment:
Note that in practice this will be called not after the wakeup, but before entering the sleep again. This is probably okay for many use cases, but it should be noted in the description of the new API.

Reply:
Oops, thanks for pointing that out. I think it would be best to notify the app via the callback as soon as possible.
I've had another look though and I'm not sure how to make that change - can you point me in the right direction?